### PR TITLE
K1J-349: Implemented qr-code and autostarttoken for eleg sign

### PR DESCRIPTION
**Pull request by mikeyOne**

### CODE REVIEW

File: apps/webcert/src/store/certificate/certificateMiddleware.ts [hunk 1]
```diff
@@ -413,38 +415,44 @@ const handleSignCertificateStatusSuccess: Middleware<Dispatch> =
...
+            apiCallBegan({
+              url: `/api/signature/${certificate.metadata.type}/${action.payload.id}/signeringsstatus`,
+              method: 'GET',
+              onSuccess: signCertificateStatusSuccess.type,
+              onError: signCertificateStatusError.type,
+              functionDisablerType: toggleCertificateFunctionDisabler.type,
+            })
+          )
+      }
+    }, 1000)
```
Comment (SamuelTilly) [inline apps/webcert/src/store/certificate/certificateMiddleware.ts:455]: Why the 1-second wait here?

**Review by SamuelTilly — APPROVED** [review]
Very nice! Please fix the linting errors and take a look at my suggestions. ✌️

File: apps/webcert/src/components/qrcode/QRCode.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (!qrImage) {
+    return null
+  }
+
+  return (
+    <>
+      <img src={qrImage} alt={'QR-kod'} />
+    </>
+  )
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply apps/webcert/src/components/qrcode/QRCode.tsx:45]: done

File: apps/webcert/src/feature/certificate/Modals/SignCertificateErrorModal.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const onTryAgain = () => {
+    setOpenErrorModal(false)
+    dispatch(startSignCertificate())
+  }
+
+  useEffect(() => {
+    if (
+      signStatus == CertificateSignStatus.FAILED ||
+      signStatus == CertificateSignStatus.UNKNOWN ||
+      signStatus === CertificateSignStatus.NO_CLIENT
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply apps/webcert/src/feature/certificate/Modals/SignCertificateErrorModal.tsx:32]: done

File: apps/webcert/src/components/qrcode/QRCode.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+const QrCode: React.FC<Props> = ({ qrCode }) => {
+  const [qrImage, setQrImage] = useState<string>()
+
+  useEffect(() => {
+    let setState = true
+    QRCode.toDataURL(qrCode, qrCodeOptions)
+      .then((url) => {
+        if (setState) {
+          setQrImage(url)
+        }
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply apps/webcert/src/components/qrcode/QRCode.tsx:25]: done still getting warning for "Promise returned from getQRCode is ignored" for getQRCode()

File: apps/webcert/src/feature/certificate/Modals/SignCertificateModal.tsx [hunk 5]
```diff
@@ -25,38 +19,48 @@ const BankIDLogo = styled.img`
...
+              <h5 className="ids-heading-h5 iu-pb-200">För att signera intyget</h5>
+              <div className="ml-5">
+                <p>1. Öppna appen för BankID.</p>
+                <p>2. Tryck på QR-symbolen i appen för BankID.</p>
+                <p>3. Rikta kameran mot QR-koden ovan.</p>
+                <p>4. Följ instruktionerna i appen för att signera.</p>
+              </div>
+            </>
+          )}
+          <></>
```
[the remark this answers is not in the document]
Comment (mikeyOne) [reply apps/webcert/src/feature/certificate/Modals/SignCertificateModal.tsx:63]: Cannot find these in the file???

File: apps/webcert/src/store/certificate/certificateMiddleware.ts (hunk 1 above)
Comment (mikeyOne) [reply apps/webcert/src/store/certificate/certificateMiddleware.ts:455]: We want to poll for changes in signstatus every second

**Pull request merged**